user-apps/windows-launcher/default.nix

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  stdenvNoCC,
+  pkgs,
+  ...
+}: let
+  windowsLauncher =
+    pkgs.writeShellScript
+    "windows-launcher"
+    ''
+      if [ $# -eq 0 ]; then
+        echo "Usage: windows-launcher ./Windows11_InsiderPreview_Client_ARM64_en-us_25324.VHDX"
+        exit
+      fi
+
+      if [[ -z "''${WAYLAND_DISPLAY}" ]]; then
+        echo "Wayland display not found"
+        exit
+      fi
+
+      IMG_DIR="$(dirname "$1")"
+      AAVMF_VARS="$IMG_DIR/AAVMF_VARS.fd"
+
+      if [ ! -f $AAVMF_VARS ]; then
+        cp ${pkgs.OVMF.fd}/FV/AAVMF_VARS.fd $AAVMF_VARS
+        chmod 644 $AAVMF_VARS
+      fi
+
+      ${pkgs.qemu}/bin/qemu-system-aarch64 \
```

Comment:
Here you could put something like

let arch = builtins.head (builtins.split "-" pkgs.stdenv.hostPlatform.system);

...

${pkgs.qemu}/bin/qemu-system-${arch}

Reply:
I think this needs even more work to make work this work on x86_64, so could we make this conditional so that this feature only appears while using NVIDIA